rules_python/whl.py

```diff
@@ -56,21 +57,25 @@ def _dist_info(self):
     return '{}-{}.dist-info'.format(self.distribution(), self.version())
 
   def metadata(self):
-    # Extract the structured data from metadata.json in the WHL's dist-info
-    # directory.
-    with zipfile.ZipFile(self.path(), 'r') as whl:
-      # first check for metadata.json
-      try:
-        with whl.open(self._dist_info() + '/metadata.json') as f:
-          return json.loads(f.read().decode("utf-8"))
-      except KeyError:
-          pass
-      # fall back to METADATA file (https://www.python.org/dev/peps/pep-0427/)
-      with whl.open(self._dist_info() + '/METADATA') as f:
-        return self._parse_metadata(f.read().decode("utf-8"))
+    if self._metadata == None:
+      # Extract the structured data from METADATA in the WHL's dist-info
+      # directory.
+      with zipfile.ZipFile(self.path(), 'r') as whl:
+        with whl.open(self._dist_info() + '/METADATA') as f:
+          # Why are we using email.parser?
+          #
+          # From PEP-0314:
+          #   The PKG-INFO file format is a single set of RFC-822 headers parseable by the rfc822.py module.
+          #   The field names listed in the following section are used as the header names.
+          #
+          # The rfc822.py module has been deprecated since version 2.3 in favor of the email package.
+          parser = email.parser.Parser()
+          self._metadata = parser.parsestr(f.read().decode('ascii', 'ignore'))
```

Comment:
I'm curious if there is a specific reason why rules_python does not want this workaround as well.